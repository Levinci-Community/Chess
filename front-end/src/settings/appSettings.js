const appSettings = {
  API_PROXY: process.env.REACT_APP_API_PROXY,
  PROXY: process.env.REACT_APP_PROXY,
  SOCKET_PROXY: process.env.REACT_APP_SOCKET_PROXY,
  OPENAI_KEY: process.env.REACT_APP_OPENAI_KEY,
  VIP_GAMES: "blitz-tactics:3000",
};

export default appSettings;
