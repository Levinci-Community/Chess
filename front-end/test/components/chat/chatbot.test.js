import React from 'react';
import renderer from 'react-test-renderer';
import { ChakraProvider } from "@chakra-ui/react";
import ChatBox from '../../../src/components/chat/chatbox';
import appSettings from '../../../src/settings/appSettings';

jest.mock('../../../src/lib/auth', () => ({
  getUserData: () => ({ id: '1', username: 'testuser' }),
}));

jest.mock('../../../src/settings/appSettings', () => ({
  OPENAI_KEY: appSettings.OPENAI_KEY,
}));

describe('ChatBox component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ChakraProvider>
        <ChatBox />
      </ChakraProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
