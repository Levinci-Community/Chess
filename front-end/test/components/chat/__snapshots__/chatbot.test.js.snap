// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBox component renders correctly 1`] = `
[
  <div
    className="css-1hiajkg"
  >
    <div
      className="css-1c9ai93"
    >
      <div
        className="css-5gbtbj"
      >
        <p
          className="chakra-text css-15wx7wh"
        >
          @
          chat.assistant
        </p>
        <p
          className="chakra-text css-153nned"
        >
          chat.hello
        </p>
      </div>
      <div
        className="css-19vstsr"
      >
        <button
          className="chakra-button css-fj8mtd"
          onClick={[Function]}
          type="button"
        >
          chat.about_ute_chess_club
        </button>
        <button
          className="chakra-button css-fj8mtd"
          onClick={[Function]}
          type="button"
        >
          chat.when_club_offline
        </button>
        <button
          className="chakra-button css-fj8mtd"
          onClick={[Function]}
          type="button"
        >
          chat.how_to_play_chess
        </button>
      </div>
    </div>
    <div
      className="chakra-stack css-1igwmid"
    >
      <input
        className="chakra-input css-mkys3m"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder="chat.type_your_message_here"
        value=""
      />
      <button
        className="chakra-button css-ez23ye"
        onClick={[Function]}
        title="chat.send"
        type="button"
      >
        <svg
          fill="currentColor"
          fontSize={24}
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m476.59 227.05-.16-.07L49.35 49.84A23.56 23.56 0 0 0 27.14 52 24.65 24.65 0 0 0 16 72.59v113.29a24 24 0 0 0 19.52 23.57l232.93 43.07a4 4 0 0 1 0 7.86L35.53 303.45A24 24 0 0 0 16 327v113.31A23.57 23.57 0 0 0 26.59 460a23.94 23.94 0 0 0 13.22 4 24.55 24.55 0 0 0 9.52-1.93L476.4 285.94l.19-.09a32 32 0 0 0 0-58.8z"
          />
        </svg>
      </button>
    </div>
  </div>,
  <span />,
  <span
    hidden={true}
    id="__chakra_env"
  />,
]
`;
